package-res/ccc/core/base/panel/legend/scene/legend-root-scene.js

```diff
@@ -181,6 +191,16 @@ def
             // New section
             if(!isLast) section = new pvc.visual.legend.LegendItemSceneSection(sections.length);
         }
+
+        function isOverflow(itemCountMax) {
+            var desiredWidth = desiredClientSize.width || Infinity;
+            var desiredHeight = desiredClientSize.height || Infinity;
+
+            return (totalItemCount > itemCountMax) ||
```

Comment:
Please separate the item count test and do it before any other calculations are performed.